test/runtests.jl

```diff
@@ -935,6 +976,12 @@ using Test
             end
         end
         @test tmp == 16.5
+
+        @inferred layout[3]
+        @inferred layout[1:3]
+        @inferred layout[:a][3]
+        @inferred layout[:b][3]
+        @inferred layout[:b][3][1]
```

Comment:
With stronger typing, some of these can be inferred, but others can't:

```julia
@inferred layout[:a][3]      # cool: these can be inferred!
@inferred layout[:b][3]
@inferred layout[:b][3][1]

@inferred layout[3][:a]      # but these fail!
@inferred layout[3][:b]
@inferred layout[3][:b][1]
```

I thought maybe adding more types to the `Record` struct (`layout[3]`) and associated functions would do it, but apparently no, or not yet.